backend/battles/helpers.py

```diff
@@ -30,6 +30,15 @@ def get_battle_result(battle):
     creator_victory_points = 0
     opponent_victory_points = 0
 
+    # check if opponent has pokemons
+    opponent_pokemons = [
+        battle.opponent_pokemon_1,
+        battle.opponent_pokemon_2,
+        battle.opponent_pokemon_3,
+    ]
+    if all(pokemon is None for pokemon in opponent_pokemons):
+        return None
```

Comment:
I feel that a better behavior here would be to raise an exception. Think about the developer experience: in case someone mistakenly calls this function while a battle is not settled then it means there's a bug in the code so we rather break than let things fails silently

Reply:
I don't think it's possible to use the exception to do what I'm doing in the templates (checking if the winner is None). There, I need to test the value of winner to render something in battle details page only if the battle is settled. I could check the opponents Pokémons, but I thought that checking the winner would be simpler. Is that possible to have something similar and also have the exception?